libquadmath: Add libquadmath to the toolchain

ClaymorePT opened the pull request:
This commit enables the build and install of the quadmath library for the GCC toolchain.
The quadmath library is required by the `charconv` Boost 1.85 library.

ClaymorePT commented:
@neheb FYI

Required for https://github.com/openwrt/packages/pull/24022

neheb commented:
I don't have merge access here.

hauke commented:
I do not find any library named `libquadmath.so` in my gcc build for aarch64 and mips32. Do we have to activate the build of  `libquadmath.so` manually with `--enable-libquadmath`?

ClaymorePT commented:
> > I do not find any library named `libquadmath.so` in my gcc build for aarch64 and mips32. Do we have to activate the build of `libquadmath.so` manually with `--enable-libquadmath`?
> 
> According to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96016, the macro __float128 will be checked even if `--enable-libquadmath` is specified.
> 
> From https://gcc.gnu.org/onlinedocs/gcc-10.2.0/gcc/Floating-Types.html#Floating-Types, it can be found that `__float128` is only supported by some targets. So we should only allow builds for x86 and PowerPC. @ClaymorePT

Fair enough. I'll make the changes.

ClaymorePT commented:
@brvphoenix In 8080a43 the makefile is now restricting to x86, x86_ 64 and apm821xx (PPC)

ClaymorePT commented:
@hauke can you try with this branch new update, please?

Ansuel commented:
@hauke This additional thing got me check the ext-toolchain script and notice we miss some librari for the FILE_SPEC configs... Notice this and ATOMIC and ASAN are missing. Should we add those there? Or it's ok?

```
	for lib in C RT PTHREAD GCC STDCPP SSP GFORTRAN GOMP; do
		local file
		local spec=""
		local llib="$(echo "$lib" | sed -e 's#.*#\L&#')"
		for file in $(find_libs "$lib"); do
			spec="${spec:+$spec }$(echo "$file" | sed -e "s#^$TOOLCHAIN#.#")"
		done
		if [ -n "$spec" ]; then
			echo "CONFIG_PACKAGE_lib${llib}=y" >> "$config"
			echo "CONFIG_LIB${lib}_FILE_SPEC=\"$spec\"" >> "$config"
		else
			echo "# CONFIG_PACKAGE_lib${llib} is not set" >> "$config"
		fi
	done
```

ClaymorePT committed:
libquadmath: Add libquadmath to the toolchain

This commit makes the libquadmath library available to the GCC
toolchain. This library is important for libraries such as
Boost.charconv

Signed-off-by: Carlos Miguel Ferreira <carlosmf.pt@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/15637
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

ClaymorePT commented:
> > @brvphoenix In [8080a43](https://github.com/openwrt/openwrt/commit/8080a43c3fe1228176452dfa5796d7ecf8358b62) the makefile is now restricting to x86, x86_ 64 and apm821xx (PPC)
> 
> I have checked and there is no `libquadmath.so` in the SDKs/toolchains for `apm821xx` and `mpc85xx`. I'm not familiar with these targets, and perhaps the `vector scalar (VSX) instruction set` is not enabled. It should be enough to add the `libquadmath` for i386 and x86_64.

Applying for x86 and x86_64 only in dfb3b4a

ClaymorePT commented:
@hauke @Ansuel @brvphoenix @neheb 

So, I see that this seems to be ready for merge. Do you guys know how we can speed this up? It seems that 6 reviews are required but the repo doesn't seem to have Ownership defined to automatically call a reviewers group. What should I do?

hauke commented:
> @hauke This additional thing got me check the ext-toolchain script and notice we miss some library for the FILE_SPEC configs... Notice this and ATOMIC and ASAN are missing. Should we add those there? Or it's ok?

Probably someone missed to adding it.
